### DISCUSSION TITLE
MAUI - XAML - Object reference not set to an instance of an object.

### DISCUSSION THREAD

**Discussion by SaddamBInSyed** [dotnet/maui#8268] [Q&A] [answered]
### Description

 
Error		Object reference not set to an instance of an object.	MauiApp1	E:\Win10\M\MobileApps\MauiApp1\MainPage.xaml	1	

### Steps to Reproduce

[IcelandMoss_MAUI.zip](https://github.com/dotnet/maui/files/8960788/IcelandMoss_MAUI.zip)

Tested in vs2022 17.2. (latest preview )

I am trying to convert this [repo ](https://github.com/kphillpotts/IcelandMoss)(https://github.com/kphillpotts/IcelandMoss) into MAUI project.

But finally, I am facing some weird issues as attached below.

 I could not able to solve this error. I am breaking my head since I am a newbie to MAUI

so please advise what I am doing wrong here ( source code attached )

Thanks 

### Version with bug

6.0 Release Candidate 2 or older

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

Android Emulator ( API 23) , Windows

### Did you find any workaround?

nope, I am a newbie.

The error, 

![image](https://user-images.githubusercontent.com/10062672/175119464-882c4111-0d18-4200-964a-3546f0d2985e.png)

### Relevant log output

```shell
Severity	Code	Description	Project	File	Line	Suppression State
Error		Object reference not set to an instance of an object.	MauiApp1	E:\Win10\M\MobileApps\MauiApp1\MainPage.xaml	1
```

**Comment by SaddamBInSyed** [comment]
@Eilon 
@jamesmontemagno 
@jfversluis 

Do I need to provide any additional information in order to find the root cause?
As informed I attached my source code as well so that you can reproduce it easily.

thanks

**Comment by AndreasReitberger** [comment, marked as answer]
@SaddamBInSyed 
This is not an issue of MAUI, but your code.
I checked your project and found the root cause.

The problem is your DataTrigger. It reference to an unkown Control `pancake:PancakeView`.

```xaml
<Border.Triggers>
        <DataTrigger
              Binding="{Binding ShoppingCart.ItemCount}"
              TargetType="pancake:PancakeView"
              Value="0">
            <Setter Property="IsVisible" Value="False" />
        </DataTrigger>
    </Border.Triggers>
```

Add a reference to "pancake:" in your XAML file header (you need to find the MAUI counterpart for this), or remove the trigger

```xaml
xmlns:pancake="clr-namespace:Xamarin.Forms.PancakeView;assembly=Xamarin.Forms.PancakeView"
```
https://github.com/kphillpotts/IcelandMoss/blob/master/src/IcelandMoss/IcelandMoss/MainPage.xaml#L11

Or, add `Border` as `TargetType`, if you want to set the parent Border control to invisible there.

Then, it will built.

**Reply by SaddamBInSyed** [reply]
@AndreasReitberger 
Thank you so much.

**Comment by jfversluis** [comment]
Hey @SaddamBInSyed thanks for your interest in .NET MAUI! I appreciate the tag, but there is no need for it. We have procedures in place to work through the incoming issues :)

Additionally, this seems like a question rather than an actual issue in .NET MAUI, if you have questions, please use either Discussions or Stack Overflow or forums like that for the more "How To" questions.

And while I was typing this, our wonderful community has stepped in and @AndreasReitberger has done an amazing job at explaining all this! Thank you so much!
